src/main/java/com/marklogic/client/helper/DatabaseClientConfig.java

```diff
@@ -4,7 +4,11 @@
 
 import com.marklogic.client.DatabaseClientFactory.Authentication;
 import com.marklogic.client.DatabaseClientFactory.SSLHostnameVerifier;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
 
+@Component
```

Comment:
I like the idea then of keeping DCC the way it is - and where it's not in a package with "spring" in it - and making a Spring-specific subclass of it that uses the constructor args and has Component on it. MarkLogicApplicationContext can then use that class.